serial_port : "/dev/roboteq" # Udev rule to make a virtual port that's associated to specific hw
baudrate : 115200
cmd_vel_topic: "cmd_vel_roboteq"

#i changed this
closed_loop: false

# For differential drive control (dual channel controller)
diff_drive_mode: false
wheel_circumference: 0.3667 # [m]
track_width: .289 # [m]
max_rpm: 122 # [rpm]

#For odometry
base_frame: "base_link"
odom_frame: "odom"
encoder_topic_name: "encoder_count"
command_srv: "/dualchannel_command_service"
reduction_ratio_internal: 0.0204    # 1/49, internal gearing ratio, see https://www.superdroidrobots.com/store/electrical-parts/encoders-accessories/encoder-motors/product=849
reduction_ratio_external: 0.3125    # 10/32, external gearing ratio
radius: 0.05839    # measured tracks
wheelbase: 0.2794  # measured
poles: 5
rate: 5
ppr: 1024
encoder_max: 65536
encoder_min: -65536

# For querying data from the motor controller
frequency: 100   #higher frequency (value is in ms)
query:
  motor_amps : ?A 
  motor_command : ?M
  # hall_count : ?Cb
  # hall_speed : ?BS
  volts : ?V
  feedback : ?F 
  # battery_amps : ?BA
  # power : ?P          
  # fault_flag : ?FF  
  # motor_amps2 : ?MA
  # motor_command2 : ?FM
  # hall_count2 : ?FM
  # hall_speed2 : ?FS
  # volts2 : ?V
  # feedback2 : ?rmp 
  # battery_amps2 : ?PHA
  # power2 : ?sns          
  # fault_flag2 : ?S  
  # status_flag : ?FS
  # firmware_id : ?FID  
  encoder_count : ?C
  encoder_speed : ?S  
